app/controllers/cloud_volume_controller.rb

```diff
@@ -24,23 +24,41 @@ def self.display_methods
     'cloud_volume_snapshot_create' => [:snapshot_create, 'snapshot_new'],
     'cloud_volume_backup_create'   => [:backup_create,   'backup_new'],
     'cloud_volume_backup_restore'  => [:backup_restore,  'backup_select'],
+    'cloud_volume_refresh'         => [nil, 'cloud_volume_refresh'],
   }.freeze
 
   def specific_buttons(pressed)
     return false unless BUTTON_TO_ACTION_MAPPING.include?(pressed)
 
-    validate_action, ui_action = BUTTON_TO_ACTION_MAPPING[pressed]
-    if validate_action
-      validate_results = validate_item_supports_action_button(validate_action, CloudVolume)
-      if validate_results[:action_supported]
-        javascript_redirect(:action => ui_action, :id => checked_item_id)
+    if pressed == "cloud_volume_refresh"
+      if @_params["miq_grid_checks"]
+        emss = Set.new
+        records = @_params["miq_grid_checks"].split(',')
+        records.each do |record|
+          emss.add(find_record_with_rbac(CloudVolume, record)&.ext_management_system)
+        end
+        emss.each { |ems| EmsRefresh.refresh(ems) }
```

Comment:
This is assuming way too much, specifically doing a full refresh when the provider might support targeted refresh of cloud volumes

Reply:
is this problem relevant only to cloud volumes or is it relevant to all other models in this PR as well?
doesn't this make a single refresh button inherently ambiguous?
maybe we could add a button specific for an ems refresh for selected records, which also makes the special casing ``if pressed == "cloud_volume_refresh" less problematic.. 
unless @jeffibm or you could find a better solution for our issue here:

the button for ems refresh already exists in the show_list and show pages of the various storage models, but it doesn't work. 

in the case of the general show_list view (not that one accessed through a specific provider), it might present records belonging to several providers (especially in the case of cloud volumes), so we want to iterate thru the emss of the selected records.

Reply:
If the refresh button is on a `CloudVolume` then we should refresh the cloud volume, not the cloud volume's EMS.  It should be up to the specific provider if they are able to do a targeted refresh or not on the specific cloud_volume instance.

Reply:
so in case that a provider can't perform a targeted refresh, the refresh button should appear disabled?

<img width="586" alt="Screenshot 2023-05-15 at 12 00 37" src="https://github.com/ManageIQ/manageiq-ui-classic/assets/106743023/5626b6cb-9467-481a-9269-6e517dd4a5bc">

our team doesn't plan currently to add an option for targeted refresh on selected records in the `show` / `show_list` views. 
we propose to add `supports_not :refresh` to our provider's models, and to change the `:klass` of the refresh button to `PolymorphicConditionalButton`, which together will disable the refresh on our models. Is this acceptable?

Reply:
@OrGur1987 I think that is still more work than it is worth.  If you just pass the the cloud_volume record to `EmsRefresh.queue_refresh(cloud_volume_1, cloud_volume_2, etc...)` everything will work the way it should.  If a provider supports targeted refresh it will refresh just those cloud_volumes and if a provider doesn't then it will run a full refresh.